src/types/task/ActionEffect.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  asArray,
+  asBoolean,
+  asEither,
+  asNumber,
+  asObject,
+  asOptional,
+  asString,
+  asValue,
+  Cleaner
+} from 'cleaners'
+
+// -------------------------------------------------------------------
+// Type definitions
+// -------------------------------------------------------------------
+
+export interface GeneralActionEffect {
+  completed: boolean
+}
+export interface SeqActionEffect extends GeneralActionEffect {
+  type: 'seq'
+  opIndex: number
+  childEffect: ActionEffect
+}
+
+export interface ParActionEffect extends GeneralActionEffect {
+  type: 'par'
+  childEffects: ActionEffect[]
+}
+
+export interface BalanceActionEffect extends GeneralActionEffect {
+  type: 'balance'
+  address: string
+  aboveAmount?: string
+  belowAmount?: string
+  walletId: string
+  tokenId?: string
+}
+
+export interface TxConfsActionEffect extends GeneralActionEffect {
+  type: 'tx-confs'
+  txId: string
+  walletId: string
+  confirmations: number
+}
```

Comment:
These are client-side types, not server types. I talked to Sam about this, and we agreed that the client needs to "bake" these down a bit more before sending them to the server.

For instance, `walletId` is useless to the server, since the server cannot access customer wallets. Instead, the client needs to extract a single network address for the server to check. Likewise, `par` and `seq` are something the client worries about as it executes the program. Once the client figures out what the next steps are, it uploads them to the server, so the server doesn't need to know how they are linked.

Nevertheless, I think we can go ahead and merge this now and fix it later - I don't want to slow down the project, and something is better than nothing.